pkg/utils/kubernetes/crd_test.go

```diff
@@ -0,0 +1,187 @@
+
+package kubernetes_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsscheme "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/scheme"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kubernetesscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	. "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("CRD", func() {
+	var (
+		ctx        context.Context
+		applier    kubernetes.Applier
+		testClient client.Client
+
+		unreadyCRD = &apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "myresources.mygroup.example.com",
+			},
+		}
+		readyCRD = &apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "myresources.mygroup.example.com",
+			},
+			Status: apiextensionsv1.CustomResourceDefinitionStatus{
+				Conditions: []apiextensionsv1.CustomResourceDefinitionCondition{
+					{Type: apiextensionsv1.Established, Status: apiextensionsv1.ConditionTrue},
+					{Type: apiextensionsv1.NamesAccepted, Status: apiextensionsv1.ConditionTrue},
+				},
+			},
+		}
+
+		validManifest   string
+		invalidManifest string
+
+		crdDeployer *CRDDeployer
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		validManifest = `apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+    name: myresources.mygroup.example.com`
+		invalidManifest = `thisIsNotAValidManifest`
+
+		testClient = fakeclient.NewClientBuilder().WithScheme(kubernetesscheme.Scheme).Build()
+		mapper := meta.NewDefaultRESTMapper([]schema.GroupVersion{apiextensionsv1.SchemeGroupVersion})
+		mapper.Add(apiextensionsv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"), meta.RESTScopeRoot)
+		applier = kubernetes.NewApplier(testClient, mapper)
+		var err error
+		crdDeployer, err = NewCRDDeployer(testClient, applier, []string{validManifest})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crdDeployer).ToNot(BeNil())
+	})
+
+	Describe("#Deploy", func() {
+		It("should deploy a CRD", func() {
+			actualCRD := &apiextensionsv1.CustomResourceDefinition{}
+
+			err := crdDeployer.Deploy(ctx)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = testClient.Get(ctx, client.ObjectKey{Name: readyCRD.Name}, actualCRD)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(actualCRD.Name).To(Equal(readyCRD.Name))
+
+		})
+	})
+
+	Describe("#Destroy", func() {
+		It("should destroy a CRD", func() {
+			actualCRD := &apiextensionsv1.CustomResourceDefinition{}
+
+			err := testClient.Create(ctx, readyCRD)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Why not in one line? Expect(...).To(Succeed())?
Please take a look at other unit tests and the style there.
Apply this for similar occurrences in this PR.